### DISCUSSION TITLE
Exit codes of 4C

### DISCUSSION THREAD

**Discussion by gilrrei** [4C-multiphysics/4C#175] [Q&A] [answered]
@queens-py / @queens-py/maintainers commonly rely on 4C as physics solver. Here, the question emerged of how 4C handles its exit codes, as we rely on them to trigger different behaviour.

Question numero 1:
Is the exit code always 1 on failure, or are there different ones?

Question numero 2:
Is it possible to exit with different codes depending on the error? In a discussion with @sebproell we came to the following relevant ones:
- Input failure (e.g., reading inputs, geometry, non-existing paths, :loudspeaker: @c-p-schmidt `electrode.cpp, line 75:
Invalid file!`, invalid parameters) 
- Solver failure (e.g., divergence of non/linear solver(s))
- Other possible failure (e.g., non-existence of files, write/read permission, mpi stuff)

For MQA, having this information is vital as a failure due to a bad parameterization solver failure might appear; however, I might still want to continue the MQA. In contrast, if the input file can not be read, the MQA should stop immediately. For the last type I think it's more case dependent.

**Comment by amgebauer** [comment, marked as answer]
> Is the exit code always 1 on failure, or are there different ones?

Currently, all exceptions that occur within 4C (and that we can catch return 1). But the process might also be killed by another action (e.g. on the cluster by slurm, or by the system, or on a segmentation fault). In that case, the exit code is typically not 1.

> Is it possible to exit with different codes depending on the error?

I think we should convert this part of the question into an issue.

**Reply by amgebauer** [reply]
I opened #234 for question 2
